### PR TITLE
write logic to check for win/tie

### PR DESCRIPTION
**Pull request by chirchir12**
* replaced inline comments with real game logic
* improved user interface of the game
* applied the DRY
* we used single responsibility rule
* covered all the edge cases.

### CODE REVIEW

File: lib/play.rb [hunk 1]
```diff
@@ -59,7 +60,7 @@ def wrong_input
   end
 
   def position_taken_error
-    board.message = "\n       cleaThat position is taken,\n" + '    choose form the available ones'
+    board.message = "\n       That position is taken,\n" + '    choose form the available ones'
     continue
```
Comment (maelfosso) [inline lib/play.rb:40]: You can't call **board** in this module.
You must move all the methods, from the `Play` module, inside which you called or used `board.` into the **Game** class directly.
The module **Play** like the one you created should only contains additional methods that must help the **Game** class.
**CHANGE THIS CONCEPTION**!

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello @chirchir12 @Teshager21 
Nice game. Thank you for the new options proposed to the user.
However, before moving on there are some changes I would like to suggest

### Board.rb

1.
https://github.com/chirchir12/TicTacToe/blob/b42f95399eacba6cdddccab6cff45f63cebf3e59/lib/board.rb#L9-L10
Why not just use a simple hash
```rb
@cells = { '1' => ' ', '2' => ' ', '3' => ' ', '4' => 
...
```
It will be simpler than manipulating array.

2. https://github.com/chirchir12/TicTacToe/blob/b42f95399eacba6cdddccab6cff45f63cebf3e59/lib/board.rb#L11
I don't think is needed. You nowhere change it, update it. Why did you create it? You should remove it.

3. 
https://github.com/chirchir12/TicTacToe/blob/b42f95399eacba6cdddccab6cff45f63cebf3e59/lib/board.rb#L15-L16
Those 2 variables are not needed. You can obtain those values from `@current_player`

4. https://github.com/chirchir12/TicTacToe/blob/b42f95399eacba6cdddccab6cff45f63cebf3e59/lib/board.rb#L22-L28
If you had implemented the `@cells` as a simple hash, then all those lines can be replaced by only 
```rb
@cells.values
```
Which would return an array of values.

5. https://github.com/chirchir12/TicTacToe/blob/b42f95399eacba6cdddccab6cff45f63cebf3e59/lib/board.rb#L38-L65
All this methods must begin with this line
```rb
values = extract_values
....
# manipulating values instead of extract_values
```
Because, the method `extract_values` cost something and in each of this method every time you want to access to one of its values (`extract_values[0]`) you make this method build the array again before returning one value. So for only one method, the method if called thrice. By doing as indicated, the method is called only once and in the next line you just use the result.

6.
https://github.com/chirchir12/TicTacToe/blob/b42f95399eacba6cdddccab6cff45f63cebf3e59/lib/board.rb#L86-L91
No `puts/gets` in `lib/`. Read [this](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png), please.

7.
https://github.com/chirchir12/TicTacToe/blob/b42f95399eacba6cdddccab6cff45f63cebf3e59/lib/board.rb#L103-L107
Not needed


### ui.rb

1. As indicated, No `puts/gets` in any files of `lib/` folder

2. As mentioned in one comment below, don't use the module concept like you are doing. You must write independent method. If one method will use an attribute from a certain class, move that method inside the class in question.

For example, the `ui` method is using `@cells`. You must move the `ui` method into the `Board` class and just called it
```rb
board.ui
```
wherever is needed

That is a mis-conception problem.

### play.rb

This file must be updated according to indications give for `ui.rb` up there.

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Comment by Teshager21** [comment]
- removed all the redundant attributes
- avoid the repeated calling of methods
- remove puts and gets from the lib folder

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team!
### You have done a good job so far :clap: However, there are a few issues to solve before your project could be approved.
- [ ] In the game, i cannot determine a win or a draw
game
### Make these changes and request for another code review. Good luck.
- Happy coding!

By Raymond
Slack @iamraymond

**Comment by chirchir12** [comment]
* Hello the rules of the game are that when there are matching Os or Xs diagonally, it is a win. that screenshot is a win

**Comment by rOluochKe** [comment]
> * Hello the rules of the game are that when there are matching Os or Xs diagonally, it is a win. that screenshot is a win

Sorry about the message that returns when one wins the game, now make it in a loop that it doesn't terminate when a player wins

File: lib/board.rb [hunk 2]
```diff
@@ -2,18 +2,15 @@
 require_relative 'player'
 class Board
   include UserInterface
-  attr_reader :option
   attr_accessor :state, :player_move, :current_player, :player1, :player2, :message, :cells, :next_player
```
Comment (MauricioRobayo) [inline lib/board.rb:5]: [OPTIONAL] Could you please double-check that all of these need to be `attr_accessor` or if you could move some of them into `attr_reader` to have better control of what others (including yourselves from the future) can do with your class ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

**Review by MauricioRobayo — CHANGES_REQUESTED** [review]
Wow, this is a nice UI, congratulations, I really like what you have done here 😄 .

Please take into account the previous feedback from @rOluochKe , the applications throws an error when the game is over, either if it is a draw or someone wins. Could you please check that so it doesn't throw an error:

image

- [ ] Fix error when the game is over.

Status: __Request changes__.

---

OPTIONAL: You can remove the `.gitkeep` file inside lib by now as the `lib` directory is no longer empty: https://www.sololearn.com/Discuss/1336098/what-are-keep-and-gitkeep-used-for-in-git

**Review by adaorachi — APPROVED** [review]
### **Hi @chirchir12 and @Teshager21** :wave: 

Good Job implementing the changes required and fixing the error in your codes. I also want to commend the effort you put into making the README content look detailed and professional. Well done!
You can go ahead and merge this branch into the base branch of your repo.

---
### **STATUS: APPROVED** :+1: 
---

- [optional] I do recommend you take some time to look into the optional requirements suggested in the previous review and implement them. This will help you in writing better codes for future projects.

Happy Coding!

**MaryAnn Chukwuka.**

**Pull request merged**